### ISSUE TITLE
code in run_squad.py line 263

### ISSUE THREAD

**Issue by hitxujian** [huggingface/pytorch-pretrained-BERT#86]
# Zero-pad up to the sequence length.
 while len(input_ids) < max_seq_length:
        input_ids.append(0)
        input_mask.append(0)
        segment_ids.append(0)
in segment_ids array,1 indicates token from passage and 0 indicate token form query.
when padding,why segment_ids filled with 0,which represents query

**Comment by hitxujian** [comment]
![image](https://user-images.githubusercontent.com/11830865/49438135-61d37c00-f7f8-11e8-8b2a-a7222bd30f0e.png)

**Comment by thomwolf** [comment]
Hi, what is your question?

**Comment by bheinzerling** [comment]
Strictly speaking, the zero-padding in segment_ids leads to ambiguous tensor entries, because 0 can mean both "first sentence" (or query in another task?) and "padding".

But in practice this isn't a problem because anything related to padding gets masked out later.
